backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/descriptor/decorator/IDocumentDecorator.java

```diff
@@ -34,8 +33,9 @@ default boolean isReadOnly(final Document document)
 		return false;
 	}
 
-	default Optional<ITranslatableString> cannotBeDeleted(final Document document)
+	@NonNull
+	default BooleanWithReason cannotBeDeleted(final Document document)
 	{
-		return Optional.empty();
+		return BooleanWithReason.FALSE;
```

Comment:
maybe it's just me, but i find it a bit twisted this "cannotBeDeleted" name.
I mean i am reading here, cannotBeDeleted returns FALSE 
which means that we are not preventing this record from deleting,
which means that we can delete.

Wondering why not having an isDeleteable method similar with isReadonly.

Anyways, this might be only my issue :)
So if u don't agree, feel free to ignore.